charts/heimdall/templates/podmonitor.yaml

```diff
@@ -0,0 +1,46 @@
+{{- $podmonitorsCRD := ( .Capabilities.APIVersions.Has "monitoring.coreos.com/v1" ) }}
+{{- $metrics := default dict .Values.metrics }}
+{{- if and
+  (ne $metrics.enabled false)
+  (eq "prometheus" ( .Values.env.OTEL_METRICS_EXPORTER | default ""))
+  (not (empty $podmonitorsCRD))
+  .Values.podMonitor.enabled
+}}
+{{- $data := dict "Release" .Release "Values" .Values "Chart" .Chart "Component" "podMonitor" }}
+apiVersion: monitoring.coreos.com/v1
+kind: PodMonitor
+metadata:
+  name: {{ include "heimdall.fullname" $data }}
+  namespace: {{ include "heimdall.namespace" $data }}
```

Comment:
Prometheus operator resources often have a configurable namespace. In some setup, it's preferable to group of these into a single namespace so Prometheus operator watch scope can be constrained. I would recommend deploying it to podMonitor.namespace or default to the release namespace.

Reply:
Should we support `namespaceSelector` then as well? Or is it already covered by the `selector` (even though it acts only on labels)?